iaas-common/src/main/java/io/github/cloudiator/persistance/PricingDomainRepository.java

```diff
@@ -0,0 +1,191 @@
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+public class PricingDomainRepository {
+    private static final PricingConverter PRICING_CONVERTER = new PricingConverter();
+    private final PricingModelRepository pricingModelRepository;
+    private final ApiDomainRepository apiDomainRepository;
+    private final OperatingSystemDomainRepository operatingSystemDomainRepository;
+    private static final Logger LOGGER = LoggerFactory.getLogger(PricingDomainRepository.class);
+
+    @Inject
+    public PricingDomainRepository(PricingModelRepository pricingModelRepository, ApiDomainRepository apiDomainRepository, OperatingSystemDomainRepository operatingSystemDomainRepository) {
+        this.pricingModelRepository = pricingModelRepository;
+        this.apiDomainRepository = apiDomainRepository;
+        this.operatingSystemDomainRepository = operatingSystemDomainRepository;
+    }
+
+    public DiscoveredPricing findById(String id) {
+        return PRICING_CONVERTER.apply(pricingModelRepository.findByCloudUniqueId(id));
+    }
+
+    public double findPrice(String cloudProviderName, String locationProviderId, String hardwareProviderId, String osArchitecture, String osFamily) throws Exception {
+        ApiModel api = apiDomainRepository.findModelByProviderName(cloudProviderName);
+
+        OperatingSystemArchitecture operatingSystemArchitecture = OperatingSystemArchitecture.valueOf(osArchitecture);
+        OperatingSystemArchitecture operatingSystemArchitectureFind;
+        switch (operatingSystemArchitecture) {
+            case AMD64:
+            case UNKNOWN: {
+                operatingSystemArchitectureFind = OperatingSystemArchitecture.AMD64;
+                break;
+            }
+            case I368: {
+                operatingSystemArchitectureFind = OperatingSystemArchitecture.I368;
+                break;
+            }
+            default: {
+                throw new Exception(String.format("No data for Operating System Architecture %s", osArchitecture));
+            }
+        }
+
+        OperatingSystemFamily operatingSystemFamily = OperatingSystemFamily.valueOf(osFamily);
+        OperatingSystemFamily operatingSystemFamilyFind;
+        switch (operatingSystemFamily) {
+            case RHEL: {
+                operatingSystemFamilyFind = OperatingSystemFamily.RHEL;
+                break;
+            }
+            case SUSE: {
+                operatingSystemFamilyFind = OperatingSystemFamily.SUSE;
+                break;
+            }
+            case WINDOWS: {
+                operatingSystemFamilyFind = OperatingSystemFamily.WINDOWS;
+                break;
+            }
+            case CENTOS:
+            case AMZN_LINUX:
+            case DEBIAN:
+            case UNKNOWN:
```

Comment:
Is it a misstake? UNKNOWN is currently treated as a UBUNTU.

Reply:
not it is not - the idea behind this is that AWS doesn't have pricing info for "UNKNOWN" OS, but the OperatingSystemFamily enum has "UNKNOWN" value, so when you want pricing for UNKNOWN OS I'm presenting you the price for VM with Linux (the cheapest one)